src/app/treasure-hunt/treasure-hunt-report/treasure-hunt-report.service.ts

```diff
@@ -196,4 +197,118 @@ export class TreasureHuntReportService {
+      electricityCO2ProjectedUse: this.getCo2EmissionsResultFromObj(energyUsage.electricityCO2SavingsData, treasureHuntResults.electricity.modifiedEnergyUsage, settings),
+      naturalGasCO2CurrentUse: this.getCo2EmissionsResultFromObj(energyUsage.naturalGasCO2SavingsData, treasureHuntResults.naturalGas.baselineEnergyUsage, settings),
+      naturalGasCO2ProjectedUse: this.getCo2EmissionsResultFromObj(energyUsage.naturalGasCO2SavingsData, treasureHuntResults.naturalGas.modifiedEnergyUsage, settings),
+      otherFuelCO2CurrentUse: this.getCo2EmissionsResultFromObj(energyUsage.otherFuelCO2SavingsData, treasureHuntResults.otherFuel.baselineEnergyUsage, settings),
+      otherFuelCO2ProjectedUse: this.getCo2EmissionsResultFromObj(energyUsage.otherFuelCO2SavingsData, treasureHuntResults.otherFuel.modifiedEnergyUsage, settings),
+      waterCO2CurrentUse: this.getCo2EmissionsResultFromNumber(energyUsage.waterCO2OutputRate, treasureHuntResults.water.baselineEnergyUsage),
+      waterCO2ProjectedUse: this.getCo2EmissionsResultFromNumber(energyUsage.waterCO2OutputRate, treasureHuntResults.water.modifiedEnergyUsage),
+      wasteWaterCO2CurrentUse: this.getCo2EmissionsResultFromNumber(energyUsage.wasteWaterCO2OutputRate, treasureHuntResults.wasteWater.baselineEnergyUsage),
+      wasteWaterCO2ProjectedUse: this.getCo2EmissionsResultFromNumber(energyUsage.wasteWaterCO2OutputRate, treasureHuntResults.wasteWater.modifiedEnergyUsage),      
+      compressedAirCO2CurrentUse: this.getCo2EmissionsResultFromNumber(energyUsage.compressedAirCO2OutputRate, treasureHuntResults.compressedAir.baselineEnergyUsage),
+      compressedAirCO2ProjectedUse: this.getCo2EmissionsResultFromNumber(energyUsage.compressedAirCO2OutputRate, treasureHuntResults.compressedAir.modifiedEnergyUsage),      
+      steamCO2CurrentUse: this.getCo2EmissionsResultFromNumber(energyUsage.steamCO2OutputRate, treasureHuntResults.steam.baselineEnergyUsage),
+      steamCO2ProjectedUse: this.getCo2EmissionsResultFromNumber(energyUsage.steamCO2OutputRate, treasureHuntResults.steam.modifiedEnergyUsage)
+    }
+    return carbonResults;
+  }
+
+  getCo2EmissionsResultFromObj(data: Co2SavingsData, electricityUsed: number, settings: Settings): number {
+    //use copy for conversion data
+    let dataCpy: Co2SavingsData = JSON.parse(JSON.stringify(data));
+    let totalEmissionsResult: number;
+    if (settings.unitsOfMeasure != 'Imperial' && data.energyType == 'fuel') {
+      let conversionHelper: number = this.convertUnitsService.value(1).from('GJ').to('MMBtu');
+      dataCpy.totalEmissionOutputRate = dataCpy.totalEmissionOutputRate / conversionHelper;
+      electricityUsed = this.convertUnitsService.value(electricityUsed).from('GJ').to('MMBtu');
+    }
+    if (dataCpy.totalEmissionOutputRate && electricityUsed) {
+      totalEmissionsResult = (dataCpy.totalEmissionOutputRate) * (electricityUsed / 1000);
+    } else {
+      totalEmissionsResult = 0;
```

Comment:
Nothing wrong here, but you can be more economical by initializing totalemissionsResult to 0 and removing the else.